### PR TITLE
tab replacement

### PR DESCRIPTION
**Pull request by NoSuck**
Tabs are illegal in JSON values.  This replaces tabs with spaces.

---
This change is [Reviewable](https://reviewable.io/reviews/qutebrowser/qutebrowser/4482)

### CODE REVIEW

File: scripts/open_url_in_instance.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # initial idea: Florian Bruhin (The-Compiler)
 # author: Thore Bödecker (foxxx0)
 
-_url="$1"
+_url="${1//	/ }"
```
Comment (jgkamat) [inline scripts/open_url_in_instance.sh]: > ./scripts/open_url_in_instance.sh: 5: ./scripts/open_url_in_instance.sh: Bad substitution

could you do this in a more portable way, if this is needed at all?

Comment (NoSuck) [reply scripts/open_url_in_instance.sh]: I see the shebang has been changed to `/bin/sh`.  😏  Just a second...

**Comment by toofar** [comment]
Tabs are illegal but the literal `\t` is allowed and would get converted back to a proper tab when it got deserialized in python. This is also just one case in a general issue with things not being quoted. For instance if you pass it a URL with a `"` in it.

I think a better solution would be to make the make the IPC server return something useful to tell the user about the issue so that they can fix their input (for example pass it to jq first). Another option is to use a proper json parser, but then you will have another interpreter overhead (unless you try to use jq if it is installed first).

**Comment by The-Compiler** [comment]
At this point this really should just be a Python script using its `json` library...

**Comment by NoSuck** [comment]
@toofar:  Correct.  This patch only solves one problem, not all problems.

@The-Compiler:  I may be able to do this.  It may require the following Python modules in order to *perfectly* mimmick the original:

* codecs
* hashlib
* json
* os
* os.path
* subprocess
* sys

That seems like a lot of imports for what used to be a tiny shell script.  On the other hand, once the wrapper script is pure Python, we could toy with the possibility of incorporating it into qutebrowser proper, which could be nice.

Are the imports acceptable?

**Comment by The-Compiler** [comment]
Anything in the Python stdlib is fine, anything in the PyQt5 module might also be okay (as long as it doesn't take too long to import). Not sure what you could use instead of `socat`, maybe Python's `socket` module can do it, or you could use `QLocalSocket` from PyQt5.

**Comment by NoSuck** [comment]
I did it.  Please see PR #4484.

**Pull request closed without merging**